### PR TITLE
Implement File.mkfifo

### PR DESCRIPTION
**Pull request by ahmadsherif**

### CODE REVIEW

File: kernel/common/file.rb [hunk 1]
```diff
@@ -1195,6 +1195,21 @@ def self.sticky?(file_name)
     return false
   end
 
+  def self.mkfifo(file_name, mode = 0666)
+    raise NotImplementedError, "mkfifo is not implemented on this platform" unless Rubinius::HAVE_MKFIFO
```
Comment (YorickPeterse) [inline kernel/common/file.rb]: I think it's better to only define this method when `mkfifo` is actually present, opposed to checking for it on every call.

Comment (ahmadsherif) [inline kernel/common/file.rb]: Won't this behavior be incompatible with MRI?

Comment (brixen) [inline kernel/common/file.rb]: MRI has a special C function that is put in the method table when the method is not implemented on that platform. We do this for FFI, but we don't have a generic way to do it yet.

For now, this is fine. We already do it in a few places and if someone is doing mkfifo in a tight loop, they probably need to refactor something.

**Comment by brixen** [comment]
Thanks!

**Pull request merged**